common/djangoapps/xblock_django/user_service.py

```diff
@@ -35,11 +36,16 @@ def __init__(self, django_user, **kwargs):
         Args:
             user_is_staff(bool): optional - whether the user is staff in the course
             anonymous_user_id(str): optional - anonymous_user_id for the user in the course
+            request_country_code(str): optional -- country code determined from the user's request IP address.
+            get_user_role(callable): optional -- function which returns the user's role in the course
         """
         super().__init__(**kwargs)
         self._django_user = django_user
         self._user_is_staff = kwargs.get('user_is_staff', False)
         self._anonymous_user_id = kwargs.get('anonymous_user_id', None)
+        self._request_country_code = kwargs.get('request_country_code', None)
+
+        self.get_user_role = kwargs.get('get_user_role', lambda: 'student')
```

Comment:
Nope.. I thought maybe it was better to call the method only when it was needed, but since all that access stuff is cached, there's no risk to do that here. Fixed with 831393e .